firewood/src/storage/linear/proposed.rs

```diff
@@ -7,39 +7,55 @@ use std::io::{Error, Read};
 use std::marker::PhantomData;
 use std::sync::Arc;
 
-use super::{LinearStore, ReadLinearStore, WriteLinearStore};
+use super::layered::{Layer, LayeredReader};
+use super::ImmutableLinearStore;
 
 /// [Proposed] is a [LinearStore] state that contains a copy of the old and new data.
 /// The P type parameter indicates the state of the linear store for it's parent,
 /// which could be a another [Proposed] or is [FileBacked](super::filebacked::FileBacked)
 /// The M type parameter indicates the mutability of the proposal, either read-write or readonly
+/// TODO update comment
 #[derive(Debug)]
-pub(crate) struct Proposed<P: ReadLinearStore, M> {
-    pub(crate) new: BTreeMap<u64, Box<[u8]>>,
-    old: BTreeMap<u64, Box<[u8]>>,
-    pub(crate) parent: Arc<LinearStore<P>>,
-    phantom: PhantomData<M>,
+pub(crate) struct Proposed<T> {
```

Comment:
nit: T -> M

I like seeing M for mutability, which is all this is

Reply:
I would literally call it `Mutability` then. Sometimes it's beneficial to deviate a little from the conventions 😉

Reply:
Proposed no longer has a generic. It implements `ReadLinearStore` and `WriteLinearStore` now